api/types/types.go

```diff
@@ -258,14 +281,15 @@ type ContainerJSONPre120 struct {
 	Config    *ContainerConfig
 }
 
+// ContainerConfig is a backcompatibility struct used in ContainerJSONPre120
 type ContainerConfig struct {
 	*runconfig.Config
 
 	// backward compatibility, they now live in HostConfig
 	Memory     int64
 	MemorySwap int64
-	CpuShares  int64
-	Cpuset     string
+	CPUShares  int64
```

Comment:
Same as above, as it is *serialized* as JSON, I think it would be better to do something like :

```go
CPUShares bool `json:"CpuShares"`
CPUSet bool `json:"CpuSet"`
```